modules/ting/ting.entities.inc

```diff
@@ -258,156 +276,193 @@ class TingEntity extends DingEntity {
-            if (stripos($url, 'http') === 0 || strpos($url, '[') === 0) {
-              // Give other modules a chance to rewrite the url.
-              drupal_alter('ting_online_url', $url, $this);
-            }
-          }
-        }
-      }
-    }
-
-    // No hasOnlineAccess found so fallback to dc:identifier.
-    if (empty($url) && isset($this->reply->record['dc:identifier']['dcterms:URI'])) {
-      $url = $this->reply->record['dc:identifier']['dcterms:URI'][0];
-      // Give ting_proxy a change to rewrite the url.
-      drupal_alter('ting_online_url', $url, $this);
-    }
-
-    return $url;
+  public function getOnline_url() {
+    return empty($this->reply) ? FALSE : $this->reply->getOnlineUrl();
   }
 
-  function getAc_source() {
-    return !empty($this->reply->record['ac:source'][''][0]) ? $this->reply->record['ac:source'][''][0] : FALSE;
+  /**
+   * Whether the material is a purely online material.
+   *
+   * @return bool
+   *   TRUE if the material can only be found online.
+   */
+  public function isOnline() {
+    return empty($this->reply) ? FALSE : $this->reply->isOnline();
   }
 
-  function getDescription() {
-    return !empty($this->reply->record['dc:description'][''][0]) ? $this->reply->record['dc:description'][''][0] : FALSE;
+  /**
+   * Returns the original source of the material.
+   *
+   * Eg. Bibliotekskatalog, Anmeldelser.
+   *
+   * @return string|FALSE
+   *   The source of the material.
+   */
+  public function getAc_source() {
+    return empty($this->reply) ? FALSE : $this->reply->getMaterialSource();
   }
 
   /**
-   * Get ISBN numbers of the object.
+   * Get an description of the material.
    *
-   * @return array
+   * @return string
+   *   The description.
    */
-  function getIsbn() {
-    $isbn = array();
-
-    // Nothing to do.
-    if (empty($this->reply->record['dc:identifier']['dkdcplus:ISBN'])) {
-      return $isbn;
-    }
-
-    // Get ISBN numbers.
-    $isbn = $this->reply->record['dc:identifier']['dkdcplus:ISBN'];
-    foreach ($isbn as $k => $number) {
-      $isbn[$k] = str_replace(array(' ', '-'), '', $number);
-    }
-    rsort($isbn);
-    return $isbn;
+  public function getDescription() {
+    return empty($this->reply) ? FALSE : $this->reply->getDescription();
   }
 
+  /**
+   * The ISBN of the material.
+   *
+   * Eg. "9780615384238"
+   *
+   * @return string[]
+   *   Zero or more ISBNs.
+   */
+  public function getIsbn() {
+    return empty($this->reply) ? FALSE : $this->reply->getIsbn();
+  }
 }
 
 /**
  * Ting collection entity.
  */
 class TingCollection extends DingEntityCollection {
+  // Default properties we want handled via magic getters in DingEntityBase.
   public $title = DingEntityBase::NULL;
   public $types = DingEntityBase::NULL;
   public $types_count = DingEntityBase::NULL;
   public $primary_object = DingEntityBase::NULL;
 
-  function __construct($id) {
+  // TODO BBS-153: Remove when port is done if we can be sure it is not used.
```

Comment:
Is this possible now?

Reply:
This may be related to the lacking migration of ting_new_materials. What do you say @danquah?